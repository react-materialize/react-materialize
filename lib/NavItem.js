'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var NavItem = function NavItem(_ref) {
  var divider = _ref.divider,
      children = _ref.children,
      _ref$href = _ref.href,
      href = _ref$href === undefined ? '' : _ref$href,
      onClick = _ref.onClick,
      props = _objectWithoutProperties(_ref, ['divider', 'children', 'href', 'onClick']);

  if (divider) return _react2.default.createElement('li', { className: 'divider' });
  var a = onClick ? _react2.default.createElement(
    'a',
    { onClick: onClick },
    children
  ) : _react2.default.createElement(
    'a',
    { href: href },
    children
  );
  return _react2.default.createElement(
    'li',
    props,
    a
  );
};

NavItem.propTypes = {
  /**
   * children can be a string value or a node
   */
  children: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.node]),
  divider: _propTypes2.default.bool,
  href: _propTypes2.default.string,
  /**
   * NavItem can have onClick. If it does have, href
   * will not be rendered
   */
  onClick: _propTypes2.default.func
};

exports.default = NavItem;