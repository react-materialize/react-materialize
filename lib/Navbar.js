'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Col = require('./Col');

var _Col2 = _interopRequireDefault(_Col);

var _Icon = require('./Icon');

var _Icon2 = _interopRequireDefault(_Icon);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Navbar = function (_Component) {
  _inherits(Navbar, _Component);

  function Navbar(props) {
    _classCallCheck(this, Navbar);

    var _this = _possibleConstructorReturn(this, (Navbar.__proto__ || Object.getPrototypeOf(Navbar)).call(this, props));

    _this.renderSideNav = _this.renderSideNav.bind(_this);
    return _this;
  }

  _createClass(Navbar, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      if (typeof $ !== 'undefined') {
        $('.button-collapse').sideNav(this.props.options);
      }
    }
  }, {
    key: 'renderSideNav',
    value: function renderSideNav() {
      return _react2.default.createElement(
        'ul',
        { id: 'nav-mobile', className: 'side-nav' },
        this.props.children
      );
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          brand = _props.brand,
          className = _props.className,
          fixed = _props.fixed,
          left = _props.left,
          right = _props.right,
          href = _props.href,
          other = _objectWithoutProperties(_props, ['brand', 'className', 'fixed', 'left', 'right', 'href']);

      delete other.options;

      var classes = {
        right: right,
        'hide-on-med-and-down': true
      };

      var brandClasses = {
        'brand-logo': true,
        right: left
      };

      var content = _react2.default.createElement(
        'nav',
        _extends({}, other, { className: className }),
        _react2.default.createElement(
          'div',
          { className: 'nav-wrapper' },
          _react2.default.createElement(
            _Col2.default,
            { s: 12 },
            _react2.default.createElement(
              'a',
              { href: href, className: (0, _classnames2.default)(brandClasses) },
              brand
            ),
            _react2.default.createElement(
              'ul',
              { className: (0, _classnames2.default)(className, classes) },
              this.props.children
            ),
            this.renderSideNav(),
            _react2.default.createElement(
              'a',
              { className: 'button-collapse', href: '#', 'data-activates': 'nav-mobile' },
              _react2.default.createElement(
                _Icon2.default,
                null,
                'view_headline'
              )
            )
          )
        )
      );

      if (fixed) {
        content = _react2.default.createElement(
          'div',
          { className: 'navbar-fixed' },
          content
        );
      }

      return content;
    }
  }]);

  return Navbar;
}(_react.Component);

Navbar.propTypes = {
  brand: _propTypes2.default.node,
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  left: _propTypes2.default.bool,
  right: _propTypes2.default.bool,
  href: _propTypes2.default.string,
  /**
   * Makes the navbar fixed
   */
  fixed: _propTypes2.default.bool,
  /**
   * Options hash for the sidenav.
   * More info: http://materializecss.com/side-nav.html#options
   */
  options: _propTypes2.default.shape({
    menuWidth: _propTypes2.default.number,
    edge: _propTypes2.default.oneOf(['left', 'right']),
    closeOnClick: _propTypes2.default.bool,
    draggable: _propTypes2.default.bool
  })
};

Navbar.defaultProps = {
  href: '/',
  options: {}
};

exports.default = Navbar;