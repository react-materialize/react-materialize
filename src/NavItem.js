import React from 'react';
import PropTypes from 'prop-types';

const NavItem = ({
  divider,
  children,
  href = '',
  ...props
}) => {
  if (divider) return <li className='divider' />;
  return (
    <li {...props}>
      <a href={href}>
        { children }
      </a>
    </li>
  );
};

NavItem.propTypes = {
  /**
   * children can be a string value or a node
   */
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node
  ]),
  divider: PropTypes.bool,
  href: PropTypes.string
};

export default NavItem;
