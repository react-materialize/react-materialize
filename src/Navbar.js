import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Col from './Col';
import Icon from './Icon';

class Navbar extends Component {
  constructor (props) {
    super(props);
    this.renderSideNav = this.renderSideNav.bind(this);
  }

  componentDidMount () {
    if (typeof $ !== 'undefined') {
      $('.button-collapse').sideNav(this.props.options);
    }
  }

  renderSideNav () {
    return (
      <ul id='nav-mobile' className='side-nav'>
        {this.props.children}
      </ul>
    );
  }

  render () {
    const {
      brand,
      className,
      fixed,
      left,
      right,
      href,
      ...other
    } = this.props;

    delete other.options;

    let classes = {
      right: right,
      'hide-on-med-and-down': true
    };

    let brandClasses = {
      'brand-logo': true,
      right: left
    };

    let content = (
      <nav {...other} className={className}>
        <div className='nav-wrapper'>
          <Col s={12}>
            <a href={href} className={cx(brandClasses)}>{brand}</a>
            <ul className={cx(className, classes)}>
              {this.props.children}
            </ul>
            {this.renderSideNav()}
            <a className='button-collapse' href='#' data-activates='nav-mobile'>
              <Icon>view_headline</Icon>
            </a>
          </Col>
        </div>
      </nav>
    );

    if (fixed) {
      content = <div className='navbar-fixed'>{content}</div>;
    }

    return content;
  }
}

Navbar.propTypes = {
  brand: PropTypes.node,
  children: PropTypes.node,
  className: PropTypes.string,
  left: PropTypes.bool,
  right: PropTypes.bool,
  href: PropTypes.string,
  /**
   * Makes the navbar fixed
   */
  fixed: PropTypes.bool,
  /**
   * Options hash for the sidenav.
   * More info: http://materializecss.com/side-nav.html#options
   */
  options: PropTypes.shape({
    menuWidth: PropTypes.number,
    edge: PropTypes.oneOf(['left', 'right']),
    closeOnClick: PropTypes.bool,
    draggable: PropTypes.bool
  })
};

Navbar.defaultProps = {
  href: '/',
  options: {}
};

export default Navbar;
