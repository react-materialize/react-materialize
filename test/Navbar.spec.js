import React from 'react';
import { shallow, mount } from 'enzyme';
import Navbar from '../src/Navbar';
import NavItem from '../src/NavItem';
import mocker from './helper/mocker';

describe('<Navbar />', () => {
  let wrapper;
  const sideNavMock = jest.fn();
  const restore = mocker('sidenav', sideNavMock);

  afterAll(() => {
    restore();
  });

  test('renders', () => {
    wrapper = shallow(
      <Navbar brand="Logo" right>
        <NavItem href="get-started.html">Getting started</NavItem>
        <NavItem href="components.html">Components</NavItem>
      </Navbar>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('nav')).toHaveLength(1);
  });

  test('places brand on right if navbar is left aligned', () => {
    wrapper = shallow(<Navbar brand="logo" left />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('a.brand-logo.right')).toHaveLength(1);
  });

  test('adds a brand node', () => {
    const brandNode = <span className="brando">I AM BRAND</span>;

    wrapper = mount(<Navbar brand={brandNode} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('a.brand-logo').contains(brandNode)).toBeTruthy();
  });

  test('can be fixed', () => {
    wrapper = shallow(<Navbar fixed />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.hasClass('navbar-fixed')).toBeTruthy();
  });
});
