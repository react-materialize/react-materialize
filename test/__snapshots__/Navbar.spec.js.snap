// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> adds a brand node 1`] = `
<Navbar
  brand={
    <span
      className="brando"
    >
      I AM BRAND
    </span>
  }
  href="/"
  options={Object {}}
>
  <nav>
    <div
      className="nav-wrapper"
    >
      <a
        className="brand-logo"
        href="/"
      >
        <span
          className="brando"
        >
          I AM BRAND
        </span>
      </a>
      <ul
        className="hide-on-med-and-down"
      />
      <ul
        className="side-nav"
        id="nav-mobile"
      />
      <a
        className="button-collapse"
        data-activates="nav-mobile"
        href="#"
      >
        <Icon>
          <i
            className="material-icons"
          >
            view_headline
          </i>
        </Icon>
      </a>
    </div>
  </nav>
</Navbar>
`;

exports[`<Navbar /> can be fixed 1`] = `
<div
  className="navbar-fixed"
>
  <nav>
    <div
      className="nav-wrapper"
    >
      <a
        className="brand-logo"
        href="/"
      />
      <ul
        className="hide-on-med-and-down"
      />
      <ul
        className="side-nav"
        id="nav-mobile"
      />
      <a
        className="button-collapse"
        data-activates="nav-mobile"
        href="#"
      >
        <Icon>
          view_headline
        </Icon>
      </a>
    </div>
  </nav>
</div>
`;

exports[`<Navbar /> places brand on right if navbar is left aligned 1`] = `
<nav>
  <div
    className="nav-wrapper"
  >
    <a
      className="brand-logo right"
      href="/"
    >
      logo
    </a>
    <ul
      className="hide-on-med-and-down"
    />
    <ul
      className="side-nav"
      id="nav-mobile"
    />
    <a
      className="button-collapse"
      data-activates="nav-mobile"
      href="#"
    >
      <Icon>
        view_headline
      </Icon>
    </a>
  </div>
</nav>
`;

exports[`<Navbar /> renders 1`] = `
<nav>
  <div
    className="nav-wrapper"
  >
    <a
      className="brand-logo"
      href="/"
    >
      Logo
    </a>
    <ul
      className="right hide-on-med-and-down"
    >
      <NavItem
        href="get-started.html"
      >
        Getting started
      </NavItem>
      <NavItem
        href="components.html"
      >
        Components
      </NavItem>
    </ul>
    <ul
      className="side-nav"
      id="nav-mobile"
    >
      <NavItem
        href="get-started.html"
      >
        Getting started
      </NavItem>
      <NavItem
        href="components.html"
      >
        Components
      </NavItem>
    </ul>
    <a
      className="button-collapse"
      data-activates="nav-mobile"
      href="#"
    >
      <Icon>
        view_headline
      </Icon>
    </a>
  </div>
</nav>
`;
